File: src/converter/plugins/convertModuleBlock.ts
import ts from "typescript";
import {createSimplePlugin} from "../plugin.js";
import {CheckCoverageService, checkCoverageServiceKey} from "./CheckCoveragePlugin.js";

export const convertModuleBlock = createSimplePlugin((node, context, render) => {
    if (!ts.isModuleBlock(node)) return null

    const checkCoverageService = context.lookupService<CheckCoverageService>(checkCoverageServiceKey)
    checkCoverageService?.cover(node)

    return node.statements
        .map(statement => render(statement))
        .join("\n")
})

File: gradle-plugin/karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/extension/plugins/convertModuleBlock.kt
package io.github.sgrishchenko.karakum.extension.plugins

import io.github.sgrishchenko.karakum.extension.createSimplePlugin
import typescript.Node
import typescript.asArray
import typescript.isModuleBlock

val convertModuleBlock = createSimplePlugin plugin@{ node: Node, context, render ->
    if (!isModuleBlock(node)) return@plugin null

    val checkCoverageService = context.lookupService<CheckCoverageService>(checkCoverageServiceKey)
    checkCoverageService?.cover(node)

    node.statements.asArray().joinToString(separator = "\n") { render(it) }
}
